packages/mui-material/src/Button/Button.js

```diff
@@ -86,7 +86,7 @@ const ButtonRoot = styled(ButtonBase, {
       ownerState.fullWidth && styles.fullWidth,
     ];
   },
-})(({ theme }) => {
+})(styled.fromTheme(theme => {
```

Comment:
Not sure how PigmentCSS works, could this mess up its extraction?

Reply:
I checked. Pigment CSS could work if the `fromTheme` is an independent function, not a member of the `styled`. Is it possible?

Reply:
Ok, found the `fromTheme` function below. This should work with Pigment CSS:

```js
import { fromTheme } from '../some-path';

styled('div')(fromTheme(({ theme }) => ({ … })))
```

Reply:
The function doesn't need to be attached but I did it to make it easy for typescript types, but I think maybe TS can handle inferring `theme`'s type just by being a `styled(...)()` argument.

As for the argument, I made it only `(theme) => ...` instead of `({ theme }) => ...` because that ensures that the function only needs to memoize `theme`. If the whole `props` bundle is included, then there's more stuff to memoize and that is more expensive. I've already sketched out what needs to happen for functions that need access to `props`: https://github.com/mui/material-ui/pull/43372/files#diff-8d1ff500648fb816e094c97e276c5ae924a36d6ab51d2f13afb5e3974e00c490R259-R323

Reply:
Agree, let's use `fromTheme(({ theme }) => ({ … })`. Can we update the types to only expect `theme` here and how TS errors if people try to access other props (that would be my only worry).

Reply:
I've updated the code to fit in those constraints, and I've settled on `memoTheme` for the name.